src/main/java/com/minecolonies/coremod/entity/ai/minimal/EntityAICitizenAvoidEntity.java

```diff
@@ -241,7 +241,7 @@ public boolean shouldExecute()
     public boolean shouldContinueExecuting()
     {
         stateMachine.tick();
-        return citizen.isCurrentlyFleeing();
+        return citizen.isCurrentlyFleeing() && !(citizen.getCitizenJobHandler().getColonyJob() instanceof AbstractJobGuard);
```

Comment:
Please turn this into a method on the job instead of a instance of.
Something like: job.requiresEnemyEvasion()